src/Symfony/Component/JsonStreamer/Mapping/Read/ValueObjectTypePropertyMetadataLoader.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonStreamer\Mapping\Read;
+
+use BcMath\Number;
+use Symfony\Component\JsonStreamer\Exception\InvalidArgumentException;
+use Symfony\Component\JsonStreamer\Mapping\PropertyMetadataLoaderInterface;
+use Symfony\Component\JsonStreamer\ValueTransformer\IntStringToBcMathNumberValueTransformer;
+use Symfony\Component\JsonStreamer\ValueTransformer\IntStringToGmpNumberValueTransformer;
+use Symfony\Component\JsonStreamer\ValueTransformer\StringToDateTimeValueTransformer;
+use Symfony\Component\TypeInfo\Type;
+use Symfony\Component\TypeInfo\Type\UnionType;
+
+/**
+ * Transforms scalar to value object.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @internal
+ */
+final class ValueObjectTypePropertyMetadataLoader implements PropertyMetadataLoaderInterface
```

Comment:
should this be split as well for the same reasons (https://github.com/symfony/symfony/pull/59915#discussion_r1981601727)?